### ISSUE TITLE
[Bug]: If an IPv6 address contains uppercase letters, the firewall rule is being recreated on each `terraform apply`

### ISSUE THREAD

**Issue by ThomasLandauer** [hetznercloud/terraform-provider-hcloud#414]
### What happened?

See above

### What did you expect to happen?

Don't recreate the rule ;-)

### Please provide a minimal working example

```
resource "hcloud_firewall" "foo" {
    name = "foo"
    rule {
        direction  = "in"
        protocol   = "tcp"
        port       = "22"
        source_ips = ["Aaaa:aaaa:aaaa:aaaa::/64"]
    }
}
```

**Comment by ThomasLandauer** [comment]
A weird side effect is that in this case (i.e. upper case letter in IPv6 address), a totally unrelated rule is being recreated too.:
```
Terraform will perform the following actions:

  # hcloud_firewall.app will be updated in-place
  ~ resource "hcloud_firewall" "foo" {
        id     = "..."
        name   = "foo"
        # (1 unchanged attribute hidden)

      - rule {
          - destination_ips = [] -> null
          - direction       = "in" -> null
          - protocol        = "icmp" -> null
          - source_ips      = [
              - "0.0.0.0/0",
              - "::/0",
            ] -> null
        }
      + rule {
          + destination_ips = []
          + direction       = "in"
          + protocol        = "icmp"
          + source_ips      = [
              + "0.0.0.0/0",
              + "::/0",
            ]
        }
        # (10 unchanged blocks hidden)
    }
```
I initially thought this was another bug, and found out that adding `port = ""` to the ICMP rule is a workaround.

**Comment by fhofherr** [comment]
Fixed in #468
